tests/css/test_stylesheet.py

```diff
@@ -7,6 +7,98 @@
 from textual.css._help_renderables import HelpText
 from textual.css.stylesheet import Stylesheet, StylesheetParseError
 from textual.css.tokenizer import TokenizeError
+from textual.dom import DOMNode
+from textual.geometry import Spacing
+
+
+def _make_stylesheet(css: str) -> Stylesheet:
+    stylesheet = Stylesheet()
+    stylesheet.source["test.css"] = css
+    stylesheet.parse()
+    return stylesheet
+
+
+def test_stylesheet_apply_highest_specificity_wins():
+    """#ids have higher specificity than .classes"""
+    css = "#id {color: red;} .class {color: blue;}"
+    stylesheet = _make_stylesheet(css)
+    node = DOMNode(classes="class", id="id")
+    stylesheet.apply(node)
+
+    assert node.styles.color == Color(255, 0, 0)
+
+
+def test_stylesheet_apply_doesnt_override_defaults():
+    css = "#id {color: red;}"
+    stylesheet = _make_stylesheet(css)
+    node = DOMNode(id="id")
+    stylesheet.apply(node)
+
+    assert node.styles.margin == Spacing.all(0)
+    assert node.styles.box_sizing == "border-box"
+
+
+def test_stylesheet_apply_highest_specificity_wins_multiple_classes():
+    """When we use two selectors containing only classes, then the selector
+    `.b.c` has greater specificity than the selector `.a`"""
+    css = ".b.c {background: blue;} .a {background: red; color: lime;}"
+    stylesheet = _make_stylesheet(css)
+    node = DOMNode(classes="a b c")
+    stylesheet.apply(node)
+
+    assert node.styles.background == Color(0, 0, 255)
```

Comment:
maybe using `COLOR_NAME_TO_RGB` would make these tests a bit easier to read? :slightly_smiling_face: 
```suggestion
    assert node.styles.background == COLOR_NAME_TO_RGB["blue"]
```

Reply:
I did have `Color.parse("ansi_blue")` before but didn't think it added enough as the RGB tuple is pretty clear :D